source/common/runtime/runtime_impl.cc

```diff
@@ -261,37 +304,67 @@ void SnapshotImpl::walkDirectory(const std::string& path, const std::string& pre
   }
 }
 
-LoaderImpl::LoaderImpl(Event::Dispatcher& dispatcher, ThreadLocal::SlotAllocator& tls,
-                       const std::string& root_symlink_path, const std::string& subdir,
-                       const std::string& override_dir, Stats::Store& store,
-                       RandomGenerator& generator, Api::OsSysCallsPtr os_sys_calls)
-    : watcher_(dispatcher.createFilesystemWatcher()), tls_(tls.allocateSlot()),
-      generator_(generator), root_path_(root_symlink_path + "/" + subdir),
+LoaderImpl::LoaderImpl(RandomGenerator& generator, ThreadLocal::SlotAllocator& tls)
+    : LoaderImpl(DoNotLoadSnapshot{}, generator, tls) {
+  loadNewSnapshot();
+}
+
+LoaderImpl::LoaderImpl(DoNotLoadSnapshot /* unused */, RandomGenerator& generator,
+                       ThreadLocal::SlotAllocator& tls)
+    : generator_(generator), tls_(tls.allocateSlot()) {}
+
+std::unique_ptr<SnapshotImpl> LoaderImpl::createNewSnapshot() {
+  return std::make_unique<SnapshotImpl>(generator_, values_);
+}
+
+void LoaderImpl::loadNewSnapshot() {
+  ThreadLocal::ThreadLocalObjectSharedPtr ptr = createNewSnapshot();
+  tls_->set([ptr_copy = ptr](Event::Dispatcher&)->ThreadLocal::ThreadLocalObjectSharedPtr {
```

Comment:
nit: as this is a shared_ptr, I think you can just capture `ptr` directly and don't need `ptr_copy` ?

Reply:
oops, yeah that was supposed to be `ptr = std::move(ptr)`, thx